src/ops/fixit.rs

```diff
@@ -85,37 +94,48 @@ fn exec(args: FixitArgs) -> CargoResult<()> {
     Ok(())
 }
 
+fn check(args: &FixitArgs) -> CargoResult<(impl Iterator<Item = CheckMessage>, Option<i32>)> {
+    let cmd = if args.clippy { "clippy" } else { "check" };
+    let command = std::process::Command::new(env!("CARGO"))
+        .args([cmd, "--message-format", "json"])
+        .args(args.check_flags.to_flags())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let buf = BufReader::new(Cursor::new(command.stdout));
+
+    Ok((
+        buf.lines()
+            .map_while(|l| l.ok())
+            .filter_map(|l| serde_json::from_str(&l).ok()),
+        command.status.code(),
+    ))
+}
+
 #[tracing::instrument(skip_all)]
-fn run_rustfix(
+#[allow(clippy::type_complexity)]
+fn collect_errors(
     args: &FixitArgs,
-    files: &mut IndexMap<String, File>,
-    current_target: &mut Option<(Target, String)>,
-    seen: &HashSet<(Target, String)>,
-) -> CargoResult<(IndexSet<String>, bool)> {
+    current_target: &mut Option<BuildUnit>,
+    seen: &HashSet<BuildUnit>,
+) -> CargoResult<(
```

Comment:
What if `collect_errors` doesn't care about `current_target`, reporting all possible fixes, all errors, and all completed builds?

Reply:
The caller would then all the information for
- Reporting completed builds
- Choosing the next target and immediately applying fixes if `current_target` has none
- Even in the future, fixing build-targets in parallel.

Reply:
Do we need `seen` here?

Its a cheap in complexity optimization to not process as much data so it could be nice.